codeworld-server/src/Util.hs

```diff
@@ -296,3 +299,53 @@ removeDirectoryIfExists dirName =
     handleExists e
         | isDoesNotExistError e = return ()
         | otherwise = throwIO e
+
+loadDumpedTree :: BuildMode -> UserId -> IO DirTree
+loadDumpedTree bm uid = do
+    let file = userProjectDir bm uid </> "tree.info"
+    exists <- doesFileExist file
+    realTree <- getDirectoryTree bm uid
+    case exists of
+        False -> return realTree
+        True -> do
+            content  <- LB.readFile file
+            return $ case decode content of
+                Nothing -> realTree
+                Just dumpedTree -> case compareTrees realTree (Dir "root" dumpedTree) of
```

Comment:
If you don't replace this entirely (though I think you should), I think you need to do something different here.  This seems to have the consequence that one unexpected file causes the entire directory tree to forget its sort order.

Reply:
Yes. Storing order in dir.info will resolve this.